lib/ansible/modules/cloud/scaleway/scaleway_images_facts.py

```diff
@@ -0,0 +1,177 @@
+      - EMEA-NL-EVS
+      - par1
+      - EMEA-FR-PAR1
+
+  commercial_type:
+    description:
+    - Commercial name of the compute node
+    required: true
+    choices:
+      - ARM64-2GB
+      - ARM64-4GB
+      - ARM64-8GB
+      - ARM64-16GB
+      - ARM64-32GB
+      - ARM64-64GB
+      - ARM64-128GB
+      - C1
+      - C2S
+      - C2M
+      - C2L
+      - START1-XS
+      - START1-S
+      - START1-M
+      - START1-L
+      - X64-15GB
+      - X64-30GB
+      - X64-60GB
+      - X64-120GB
+
+  timeout:
+    description:
+    - Timeout for API calls
+    required: false
+    default: 30
+'''
+
+EXAMPLES = '''
+- name: Get image information
+  scaleway_images_facts:
+    name: Ubuntu Bionic
+    region: par1
+    commercial_type: START1
+'''
+
+RETURN = '''
+data:
+    description: This is only present when C(state=present)
+    returned: when C(state=present)
+    type: dict
+    sample: {
+        "images": [
+            {
+                "arch": "x86_64",
+                "creation_date": "2018-07-26T13:00:02.726274+00:00",
+                "default_bootscript": {
+                    "architecture": "x86_64",
+                    "bootcmdargs": "LINUX_COMMON scaleway boot=local nbd.max_part=16",
+                    "default": false,
+                    "dtb": "",
+                    "id": "54ee2857-8ffb-4323-abba-964f55fea4a2",
+                    "initrd": "http://169.254.42.24/initrd/initrd-Linux-x86_64-v3.14.5.gz",
+                    "kernel": "http://169.254.42.24/kernel/x86_64-mainline-lts-4.9-4.9.93-rev1/vmlinuz-4.9.93",
+                    "organization": "11111111-1111-4111-8111-111111111111",
+                    "public": true,
+                    "title": "x86_64 mainline 4.9.93 rev1"
+                },
+                "extra_volumes": [],
+                "from_server": null,
+                "id": "67dcb730-fc17-421c-a328-f67438c74bd6",
+                "modification_date": "2018-07-26T13:00:02.726274+00:00",
+                "name": "Ubuntu Bionic Beaver",
+                "organization": "51b656e3-4865-41e8-adbc-0c45bdd780db",
+                "public": true,
+                "root_volume": {
+                    "id": "2fef7fe5-7201-487d-9cc1-058b2e6e9a12",
+                    "name": "snapshot-ea04f81d-b0bd-499f-b23a-887d01f14eb6-2018-07-25_21:28",
+                    "size": 50000000000,
+                    "volume_type": "l_ssd"
+                },
+                "state": "available"
+            }
+        ]
+    }
+'''
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible.module_utils.scaleway import ScalewayAPI, SCALEWAY_LOCATION, SCALEWAY_COMMERCIAL_TYPES_ARCH_MAPPING
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+
+
+def core(module):
+    api_token = module.params['oauth_token']
+    region = module.params["region"]
+    name = module.params["name"]
+    arch = SCALEWAY_COMMERCIAL_TYPES_ARCH_MAPPING[module.params["commercial_type"]]
```

Comment:
1. I am not sure why `commercial_type` is a required parameter since Scaleway API doesn't require `arch` parameter.
2. Why don't use directly `arch` module parameter instead of `commercial_type` parameter ? `arch` seems easiest to maintain (i guess a new commercial type is more likely to be added than a new arch).

Reply:
Maybe users won't know/care about the arch used behind a particular commercial offer and would prefer to use that. From a customer perspective, knowing the arch is an extra step compared to knowing what commercial offer they use.
On the maintenance point of view, it is essentially adding a mapping between the name and the actual arch. Not very time-consuming.

Reply:
If a new commercial type is added, Ansible users won't be able to use it, they will need to wait for an update of the code. If this addition occurs during code freeze or n-1/n-2 versions is used, this update will never happen.

Commercial type value should not be checked before calling the API. When the API returns an error, known commercial types might be displayed. `ec2_instance` module [doesn't check](https://github.com/ansible/ansible/blob/84bedcbd82a1e1cafd758eb20f18bc2542bfc106/lib/ansible/modules/cloud/amazon/ec2_instance.py#L1432, neither does `rax` module) commercial type.